specification/dashboard/resource-manager/Microsoft.Dashboard/preview/2022-10-01-preview/grafana.json

```diff
@@ -0,0 +1,1998 @@
+      "type": "object",
+      "properties": {
+        "pluginId": {
+          "description": "Grafana plugin id",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "Grafana plugin display name",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "GrafanaAvailablePluginListResponse": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/GrafanaAvailablePlugin"
+          },
+          "x-ms-identifiers": [
+            "pluginId"
+          ]
+        },
+        "nextLink": {
+          "type": "string"
+        }
+      }
+    },
+    "SaasSubscriptionDetails": {
+      "description": "SaaS subscription details of a Grafana instance",
+      "type": "object",
+      "properties": {
+        "planId": {
+          "type": "string"
+        },
+        "offerId": {
+          "type": "string"
+        },
+        "publisherId": {
+          "type": "string"
+        },
+        "term": {
+          "$ref": "#/definitions/SubscriptionTerm"
+        }
+      }
+    },
+    "SubscriptionTerm": {
+      "description": "The current billing term of the SaaS Subscription.",
+      "type": "object",
+      "properties": {
+        "termUnit": {
+          "type": "string"
+        },
+        "startDate": {
+          "description": "The date and time in UTC of when the billing term starts.",
+          "type": "string",
+          "format": "date-time"
+        },
+        "endDate": {
+          "description": "The date and time in UTC of when the billing term ends.",
+          "type": "string",
+          "format": "date-time"
+        }
+      }
+    },
+    "MarketplaceTrialQuota": {
+      "description": "The allocation details of the per subscription free trial slot of the subscription.",
+      "type": "object",
+      "properties": {
+        "availablePromotion": {
+          "description": "Available enterprise promotion for the subscription",
+          "$ref": "#/definitions/AvailablePromotion"
+        },
+        "grafanaResourceId": {
+          "description": "Resource Id of the Grafana resource which is doing the trial.",
+          "type": "string"
+        },
+        "trialStartAt": {
+          "description": "The date and time in UTC of when the trial starts.",
+          "type": "string",
+          "format": "date-time"
+        },
+        "trialEndAt": {
+          "description": "The date and time in UTC of when the trial ends.",
+          "type": "string",
+          "format": "date-time"
+        }
+      }
+    },
+    "AvailablePromotion": {
+      "enum": [
+        "None",
+        "FreeTrial"
+      ],
+      "type": "string",
+      "default": "None",
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "AvailablePromotion"
+      }
+    },
+    "ManagedPrivateEndpointModel": {
```

Comment:
There is common type available for Private Endpoint Model here: [https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/privatelinks.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/privatelinks.json). Use this one instead of defining it here.

Reply:
[must fix]

Reply:
From resource perspective, Managed Private Endpoint resource is not a Private Endpoint resource, although they share some common property names because we (RP) create private link for customer internally. But eventually what we'd like to expose can be the same or different to Private Endpoint/Link resource, also we'd like to have flexibilities like hiding some properties if we want to.

Reply:
Will not fix